### PR TITLE
925953 - [WinUI Kanban] Added kanban event changes

### PR DESCRIPTION
**Pull request by Kerubananthan3761**
## Description

WinUI Kanban control  event UG content changes added.

### CODE REVIEW

File: winui/Kanban/events.md [hunk 1]
```diff
@@ -0,0 +1,762 @@
...
+documentation: ug
+---
+
+# Events in WinUI Kanban Control
+
+## Notifying Kanban card drag start
+
+The `CardDragStarting` event occurs when the card start to drag. We can get the below details from the `CardDragStarting` event.
+
+* Column -  Returns the source column from which the card is being dragged.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:16]: Add API links for all properties

File: winui/Kanban/events.md [hunk 2]
```diff
@@ -0,0 +1,762 @@
+---
+layout: post
+title: Events in WinUI Kanban control | Syncfusion
+description: Learn here all about the interaction-related events support in Syncfusion WinUI Kanban (SfKanban) control.
+platform: winui
+control: SfKanban
+documentation: ug
+---
+
+# Events in WinUI Kanban Control
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:10]: Events in WinUI Kanban (SfKanban)

File: winui/Kanban/events.md [hunk 3]
```diff
@@ -0,0 +1,762 @@
...
+
+## Notifying Kanban card drag start
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:12]: Change to `Notifying Kanban card drag starts`

File: winui/Kanban/events.md [hunk 4]
```diff
@@ -0,0 +1,762 @@
...
+
+The `CardDragStarting` event occurs when the card start to drag. We can get the below details from the `CardDragStarting` event.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:14]: The `CardDragStarting` event occurs when a card begins to be dragged. We can get the below details from the `CardDragStarting` event.

File: winui/Kanban/events.md [hunk 5]
```diff
@@ -0,0 +1,762 @@
...
+    #endregion
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Notifying Kanban card drag over 
+
+The `CardDragOver` event occurs when card drag started in a column. We can get the below details from the `CardDragOver` event.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:148]: The `CardDragOver` event occurs when a card is being dragged over a column. We can get the following details from the `CardDragOver` event.

File: winui/Kanban/events.md [hunk 6]
```diff
@@ -0,0 +1,762 @@
...
+
+* Column - Returns the column in which the card is currently being dragged.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:150]: Add API links

File: winui/Kanban/events.md [hunk 7]
```diff
@@ -0,0 +1,762 @@
...
+* CurrentIndex - Returns the current index position of the card within its column.
+* CurrentColumnIndex - Returns the current index of the column that contains the card.
+* CurrentRowIndex - Returns the current row index of the swimlane that contains the card. The current row index is relevant only for the swimlane column and is used to get the swimlane's index.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:153]: The current row index is applicable only for the swimlane column and is used to get the swimlane's index.

File: winui/Kanban/events.md (hunk 3 above)
Comment (Karthickmani97) [inline winui/Kanban/events.md:12]: DragStarting

File: winui/Kanban/events.md [hunk 8]
```diff
@@ -0,0 +1,762 @@
...
+    }
+
+    #endregion
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Notifying Kanban card drag over
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:146]: DragOver

File: winui/Kanban/events.md [hunk 9]
```diff
@@ -0,0 +1,762 @@
...
+    }
+
+    #endregion
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Notifying Kanban card drop
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:284]: Change to `Drop`

File: winui/Kanban/events.md [hunk 10]
```diff
@@ -0,0 +1,762 @@
...
+
+The `CardDrop` event occurs when leaving the dragging card from one column. We can get the below details from the `CardDrop` event.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:286]: The `CardDrop` event occurs when a dragged card is dropped into a column. We can get the following details from the `CardDrop` event.

File: winui/Kanban/events.md [hunk 11]
```diff
@@ -0,0 +1,762 @@
...
+
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:287]: Add API links

File: winui/Kanban/events.md [hunk 12]
```diff
@@ -0,0 +1,762 @@
...
+    }
+
+    #endregion
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Notifying Kanban card tap
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:432]: Change to `CardTapped`

File: winui/Kanban/events.md [hunk 13]
```diff
@@ -0,0 +1,762 @@
...
+
+The `CardTapped` event occurs when the card is tapped. We can get the below details from the `CardTapped` event.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:434]: The `CardTapped` event occurs when the card is tapped. We can get the following details from the `CardTapped` event.

File: winui/Kanban/events.md [hunk 14]
```diff
@@ -0,0 +1,762 @@
...
+    }
+
+    #endregion
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+#### Command
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:568]: Command

Kanban commands allow to map `CardTapped` event to command which supports the MVVM (Model-View-ViewModel) pattern.

CardTappedCommand

The `CardTappedCommand` will be triggered when you tap the card and pass the `KanbanCardTappedEventArgs` as parameter.

File: winui/Kanban/events.md [hunk 15]
```diff
@@ -0,0 +1,762 @@
...
+
+        #region Constructor
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ViewModel"/> class.
+        /// </summary>
+        public ViewModel()
+        {
+            this.TaskDetails = this.GetTaskDetails();
+            CardTappedCommand = new RelayCommand(OnCardTapped, CanExecuteCardTapped);
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:611]: Use prefix of this - `this.CardTappedCommand = new RelayCommand(OnCardTapped, CanExecuteCardTapped);`

File: winui/Kanban/events.md [hunk 16]
```diff
@@ -0,0 +1,762 @@
...
+            };
+
+            taskDetails.Add(taskDetail);
+            return taskDetails;
+        }
+
+        /// <summary>
+        /// Handles the event when a Kanban card is tapped.
+        /// </summary>
+        /// <param name="parameter"></param>
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:702]: /// <summary>
        /// Handles the event when a Kanban card is tapped.
        /// </summary>
        /// <param name="parameter">The object.</param>

File: winui/Kanban/events.md [hunk 17]
```diff
@@ -0,0 +1,762 @@
...
+        private void OnCardTapped(object parameter)
+        {
+            if (parameter is KanbanCardTappedEventArgs args)
+            {
+                var selectedCoumn = args.SelectedColumn;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether card tapped can be executed.
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:712]: /// <summary>
/// Determines whether the `CardTapped` command can be executed.
/// </summary>
/// <param name="sender">The source of the command.</param>
/// <returns>Returns a boolean indicating whether the command can be executed.</returns>

File: winui/Kanban/events.md (hunk 15 above)
Comment (Karthickmani97) [inline winui/Kanban/events.md:611]: Highlight this line

File: winui/Kanban/events.md [hunk 18]
```diff
@@ -0,0 +1,762 @@
...
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <returns></returns>
+        private bool CanExecuteCardTapped(object sender)
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:716]: Highlight this methods

File: winui/Kanban/events.md [hunk 19]
```diff
@@ -0,0 +1,762 @@
...
+
+            taskDetails.Add(taskDetail);
+            return taskDetails;
+        }
+
+        /// <summary>
+        /// Handles the event when a Kanban card is tapped.
+        /// </summary>
+        /// <param name="parameter"></param>
+        private void OnCardTapped(object parameter)
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:703]: Highlight this method

File: winui/Kanban/events.md [hunk 20]
```diff
@@ -0,0 +1,762 @@
...
+    {
+        this.execute = executeAction;
+    }
+
+    public bool CanExecute(object parameter) => canExecute == null || canExecute(parameter);
+
+    public void Execute(object parameter) => execute(parameter);
+
+
+    public void RaiseCanExecuteChanged()
```
Comment (Karthickmani97) [inline winui/Kanban/events.md:752]: Have you tested this code? is it working properly?

File: winui/Kanban/events.md [hunk 21]
```diff
@@ -0,0 +1,762 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Notifying Kanban card drop
+
+The `CardDrop` event occurs when leaving the dragging card from one column. We can get the below details from the `CardDrop` event.
+
+* SourceColumn - Returns the target column out of which the user drags the card.
```
[the remark this answers is not in the document]
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Changed content.

File: winui/Kanban/events.md (hunk 20 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md:756]: yes

File: winui/Kanban/events.md (hunk 19 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md:707]: method hilighted

File: winui/Kanban/events.md (hunk 18 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md:720]: method hilighted

File: winui/Kanban/events.md (hunk 15 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: highlighted the line

File: winui/Kanban/events.md (hunk 17 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Added parm and return text.

File: winui/Kanban/events.md (hunk 16 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Added parm text.

File: winui/Kanban/events.md (hunk 15 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: added this.

File: winui/Kanban/events.md (hunk 14 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Heading and content changed

File: winui/Kanban/events.md (hunk 13 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Content changed.

File: winui/Kanban/events.md (hunk 12 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Title changed.

File: winui/Kanban/events.md (hunk 11 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md:287]: API links added.

File: winui/Kanban/events.md (hunk 10 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Content changed

File: winui/Kanban/events.md (hunk 9 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Title changed

File: winui/Kanban/events.md (hunk 8 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Title changed

File: winui/Kanban/events.md (hunk 3 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Title changed

File: winui/Kanban/events.md (hunk 7 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Content changed

File: winui/Kanban/events.md (hunk 6 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: API links added

File: winui/Kanban/events.md (hunk 5 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Content changed

File: winui/Kanban/events.md [hunk 22]
```diff
@@ -0,0 +1,762 @@
...
+{% endhighlight %}
+
+{% highlight C# hl_lines="2" %}
+
+this.kanban.ItemsSource = new ViewModel().TaskDetails;
+this.kanban.CardDragStarting += this.OnKanbanCardDragStarting;
+
+private void OnKanbanCardDragStarting(object sender, KanbanCardDragStartingEventArgs e)
+{
+    var selectedCard = e.Card;
```
[the remark this answers is not in the document]
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: var names changed

File: winui/Kanban/events.md (hunk 4 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Content changed

File: winui/Kanban/events.md (hunk 2 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Title changed

File: winui/Kanban/events.md (hunk 1 above)
Comment (Kerubananthan3761) [reply winui/Kanban/events.md]: Added API links

File: winui-toc.html [hunk 23]
```diff
@@ -377,6 +377,7 @@
             <li><a href="/winui/Kanban/getting-started">Getting Started</a></li>
             <li><a href="/winui/Kanban/columns">Columns</a></li>
             <li><a href="/winui/Kanban/cards">Cards</a></li>
+            <li><a href="/winui/Kanban/events">Events</a></li>
```
[the remark this answers is not in the document]
Comment (Kerubananthan3761) [reply winui-toc.html:381]: resolved conflicts and updated toc file

**Review by Karthickmani97 — APPROVED** [review]

**Review by Karthickmani97 — APPROVED** [review]

**Pull request merged**